docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -0,0 +1,1946 @@
+cannot change at all, only its value. Should you need to change these
+values, be aware that if their runtime value changes in any way, the
+upgrade will fail, and thus the full transaction. Contracts in this
+state can then only be used by explicitly choosing the older version of
+the contract in your transaction.
+
+Migration
+---------
+
+SCU is only supported on LF1.16, which in turn is only supported on
+Canton Protocol Version 6. This means that existing deployed contracts
+will first require migration and redeployment to utilize this feature.
+
+First, you’ll need to migrate your daml model to be compatible with
+upgrades, see the section above about `Best Practices <#best-practices>`__ for what to
+change here. Pay particular attention to the case on interfaces and
+exceptions, and failure to do so could lead to packages which are
+incompatible with SCU, and require the use of a separate tool (and
+downtime).
+
+Next, you’ll need to be aware of the new package-name scoping rules, and
+ensure that your package set does not violate this. In short, packages
+with the same package-name are considered to be “in the same scope”,
+each version string within this scope can only be held by one package,
+and contract key queries will yield contracts from all packages within
+the same scope. If you have packages with the same name that you do not
+want to be associated with each other via upgrades, you’ll need to
+rename them. Note that this scoping rule *only* applies to packages in
+LF1.16. Packages with the same name and version can exist in LF1.15, if
+you intend to use both LF versions at the same time on your participant.
+
+Once you have your new dars, you’ll need to upgrade your Canton and
+protocol version together, since 2.9 introduces a new protocol version.
+The steps to achieve this are given in the :ref:`Canton Upgrading
+manual <one_step_migration>`.
+
+Finally, you can migrate your live data from your previous dars to the
+new LF1.16 dars, using one of the existing downtime upgrade techniques
+listed in `What are Zero Downtime Smart Contract Upgrades <#what-are-zero-downtime-smart-contract-upgrades>`__.
+
+The Upgrade Model in Depth - Reference
+--------------------------------------
+
+You can find the in-depth upgrading model, which can be used as a reference
+for valid upgrades, :ref:`here <upgrade_model_reference>`.
+
+Components
+==========
+
+Ledger API
+----------
+
+Until the introduction of SCU, templates in requests on the Ledger API
+could only be referenced by the template-id, with the template fully
+qualified name of format ``<package-id>:<module-name>:<template-name>``.
+
+With SCU, we introduce a more generic template reference of the format
+``#<package-name>:<module-name>:<template-name>``. This format is only a
+Ledger API concept and is meant to suggest to the Ledger API to perform
+a dynamic runtime resolution of packages in the Daml engine when
+generating the Daml transaction after command interpretation. This
```

Comment:
```suggestion
generating the Daml transaction before command interpretation. This
```
The resolution needs to happen before command interpretation, as otherwise it's not clear what code to run.

Reply:
@tudor-da

Reply:
I see now that it reads confusing 👍